srtcore/buffer_tools.h

```diff
@@ -188,14 +190,47 @@ class CSndRateEstimator
         bool empty() const { return m_iPktsCount == 0; }
     };
 
+private:
+    static const int NUM_PERIODS        = 10;
+    static const int SAMPLE_DURATION_MS = 100; // 100 ms
+    Sample           m_Samples[NUM_PERIODS];
+    int              m_iFirstSampleIdx; //< Index of the first sample.
+
     int incSampleIdx(int val, int inc = 1) const;
+};
+
+class CMovingRateEstimator : CSndRateEstimator
+{
+    typedef sync::steady_clock::time_point time_point;
 
-    Sample m_Samples[NUM_PERIODS];
+public:
+    CMovingRateEstimator();
 
-    time_point m_tsFirstSampleTime; //< Start time of the first sample.
-    int        m_iFirstSampleIdx;   //< Index of the first sample.
-    int        m_iCurSampleIdx;     //< Index of the current sample being collected.
-    int        m_iRateBps;          //< Rate in Bytes/sec.
+    /// Add sample.
+    /// @param [in] pkts   number of packets in the sample.
+    /// @param [in] bytes  number of payload bytes in the sample.
+    void addSample(int pkts = 0, double bytes = 0);
+
+    /// Clean the mobile measures table to reset average value.
+    void resetRate() { resetRate(0, NUM_PERIODS); };
+
+    /// Retrieve estimated bitrate in bytes per second with 16-byte packet header.
+    int getRate() const { return m_iRateBps; }
+
+private:
+    const int               NUM_PERIODS        = 100; // To get 1s of values
+    const int               SAMPLE_DURATION_MS = 10;  // 10 ms
+    time_point              lastSlotTimestamp;        // Used to compute the delta between 2 calls
+    srt::FixedArray<Sample> m_Samples;                // Table of stored data
```

Comment:
Just move this to the base class instead of its `Sample m_Samples[NUM_PERIODS];`

Reply:
Done.